openpype/hosts/nuke/plugins/publish/extract_thumbnail.py

```diff
@@ -166,26 +168,45 @@ def render_thumbnail(self, instance, output_name=None, **kwargs):
             previous_node = dag_node
             temporary_nodes.append(dag_node)
 
+        thumb_name = "thumbnail"
+        # only add output name and
+        # if there are more than one bake preset
+        if (
+            output_name
+            and len(instance.data.get("bakePresets", {}).keys()) > 1
+        ):
+            thumb_name = "{}_{}".format(output_name, thumb_name)
+
         # create write node
         write_node = nuke.createNode("Write")
-        file = fhead[:-1] + output_name + ".jpg"
-        name = "thumbnail"
-        path = os.path.join(staging_dir, file).replace("\\", "/")
-        instance.data["thumbnail"] = path
-        write_node["file"].setValue(path)
+        file = fhead[:-1] + thumb_name + ".jpg"
+        thumb_path = os.path.join(staging_dir, file).replace("\\", "/")
+
+        # add thumbnail to cleanup
+        if not instance.context.data.get("cleanupFullPaths"):
+            instance.context.data["cleanupFullPaths"] = []
+        instance.context.data["cleanupFullPaths"].append(thumb_path)
+
+        # make sure only one thumbnail path is set
+        # and it is existing file
+        if instance.data.get("thumbnail"):
```

Comment:
Not sure why you're using `"thumbnail"`? Thumbnail integrator is lookin for `"thumbnailPath"`.

Reply:
Fxed! So key has been always wrong long time. Now it is `thumbnailPath` as it suppose to be.